src/Kernel32/Kernel32.cs

```diff
@@ -542,6 +542,102 @@ public static partial class Kernel32
+        /// the specified version, so the testing stops.)
+        /// </para>
+        /// <para>
+        /// To verify a range of system versions, you must call <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/> twice.For example, to verify
+        /// that the system version is greater than 5.0 but less than or equal to 5.1, first call <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/> to
+        /// test that the major version is 5 and the minor version is greater than 0, then call <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/>
+        /// again to test that the major version is 5 and the minor version is less than or equal to 1.
+        /// </para>
+        /// <para>
+        /// Identifying the current operating system is usually not the best way to determine whether a particular operating system feature is present.
+        /// This is because the operating system may have had new features added in a redistributable DLL. Rather than using GetVersionEx to determine the operating
+        /// system platform or version number, test for the presence of the feature itself.
+        /// </para>
+        /// <para>
+        /// To verify whether the current operating system is either the Media Center or Tablet PC version of Windows, call GetSystemMetrics.
+        /// </para>
+        /// <para>
+        /// Windows 10:
+        ///     <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/> returns false when called by applications that do not have a
+        ///     compatibility manifest for Windows 8.1 or Windows 10 if the <paramref name="lpVersionInformation"/> parameter is set so that it specifies
+        ///     Windows 8.1 or Windows 10, even when the current operating system version is Windows 8.1 or Windows 10. Specifically,
+        ///     <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/> has the following behavior:
+        ///
+        ///     If the application has no manifest, <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/> behaves as
+        ///         if the operation system version is Windows 8 (6.2).
+        ///     If the application has a manifest that contains the GUID that corresponds to Windows 8.1, <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/>
+        ///         behaves as if the operation system version is Windows 8.1 (6.3).
+        ///     If the application has a manifest that contains the GUID that corresponds to Windows 10, <see cref="VerifyVersionInfo(ref OSVERSIONINFOEX, VER_MASK, long)"/>
+        ///         behaves as if the operation system version is Windows 10 (10.0).
+        /// </para>
+        /// </remarks>
+        [DllImport(nameof(Kernel32))]
+        public static extern NTSTATUS VerifyVersionInfo(ref OSVERSIONINFOEX lpVersionInformation, VER_MASK dwTypeMask, long dwlConditionMask);
```

Comment:
We should use a native pointer for `lpVersionInformation` instead of `ref`. Then use the `[Friendly(FriendlyFlags.Bidirectional)]` attribute so that more friendly overloads like your `ref` here are automatically generated.

Example:

https://github.com/AArnott/pinvoke/blob/a561cd6ca12e2190db01c89e5a43ba7b3e1f71a2/src/Kernel32/storebanned/Kernel32.cs#L858